code/modules/mob/living/carbon/human/blood.dm

```diff
@@ -1,318 +0,0 @@
-			"real_name" = null,
-			"cloneable" = null,
-			"features" = null,
-			"factions" = null)
-
-/mob/living/carbon/human/proc/suppress_bloodloss(amount)
-	if(bleedsuppress)
-		return
-	else
-		bleedsuppress = 1
-		spawn(amount)
-			bleedsuppress = 0
-			if(stat != DEAD && blood_max)
-				src << "<span class='warning'>The blood soaks through your bandage.</span>"
-
-// Takes care blood loss and regeneration
-/mob/living/carbon/human/handle_blood()
-
-	if(NOBLOOD in dna.species.specflags)
-		blood_max = 0
-		return
-
-	if(stat != DEAD && bodytemperature >= 225) // Dead or cryosleep people do not pump the blood.
-		var/blood_volume = round(vessel.get_reagent_amount("blood"))
-
-		//Blood regeneration if there is some space
-		if(blood_volume < BLOOD_VOLUME_NORMAL)
-			var/datum/reagent/blood/B = locate() in vessel.reagent_list //Grab some blood
-			if(B) // Make sure there's some blood at all
-				if(B.data["donor"] != src) //If it's not theirs, then we look for theirs
-					for(var/datum/reagent/blood/D in vessel.reagent_list)
-						if(D.data["donor"] == src)
-							B = D
-							break
-
-				B.volume += 0.1 // regenerate blood VERY slowly
-				if(reagents.has_reagent("nutriment"))	//Getting food speeds it up
-					B.volume += 0.4
-					reagents.remove_reagent("nutriment", 0.1)
-				if(reagents.has_reagent("iron"))	//Hematogen candy anyone?
-					B.volume += 0.4
-					reagents.remove_reagent("iron", 0.1)
-
-		//Effects of bloodloss
-		switch(blood_volume)
-			if(BLOOD_VOLUME_SAFE to 10000)
-				if(pale)
-					pale = 0
-					update_body()
-			if(BLOOD_VOLUME_OKAY to BLOOD_VOLUME_SAFE)
-				if(!pale)
-					pale = 1
-					update_body()
-					var/word = pick("dizzy","woozy","faint")
-					src << "<span class='warning'>You feel [word].</span>"
-				adjustOxyLoss((BLOOD_VOLUME_NORMAL - blood_volume) / 100)
-			if(BLOOD_VOLUME_BAD to BLOOD_VOLUME_OKAY)
-				if(!pale)
-					pale = 1
-					update_body()
-				adjustOxyLoss((BLOOD_VOLUME_NORMAL - blood_volume) / 50)
-				if(prob(5))
-					blur_eyes(6)
-					var/word = pick("dizzy","woozy","faint")
-					src << "<span class='warning'>You feel very [word].</span>"
-			if(BLOOD_VOLUME_SURVIVE to BLOOD_VOLUME_BAD)
-				oxyloss += 5
-				if(prob(15))
-					Paralyse(rand(1,3))
-					var/word = pick("dizzy","woozy","faint")
-					src << "<span class='warning'>You feel extremely [word].</span>"
-			if(0 to BLOOD_VOLUME_SURVIVE)
-				death()
-
-		//Bleeding out
-		blood_max = 0
-		for(var/X in bodyparts)
-			var/obj/item/bodypart/BP = X
-			var/brutedamage = BP.brute_dam
-
-			//We want an accurate reading of .len
-			listclearnulls(BP.embedded_objects)
-			blood_max += 0.5*BP.embedded_objects.len
-
-			if(brutedamage > 30)
-				blood_max += 0.5
-			if(brutedamage > 50)
-				blood_max += 1
-			if(brutedamage > 70)
-				blood_max += 2
-		if(bleedsuppress)
-			blood_max = 0
-		if(reagents.has_reagent("heparin") && getBruteLoss()) //Heparin is a powerful toxin that causes bleeding
-			blood_max += 3
```

Comment:
no, we don't check for specific reagents every time we call Life(), this is stupid. I've moved the code into the on_mob_life() of heparin, nutriment, vitamin, and iron.